code/modules/power/collector.dm

```diff
@@ -0,0 +1,232 @@
+//This file was auto-corrected by findeclaration.exe on 25.5.2012 20:42:33
+var/global/list/rad_collectors = list()
+
+/obj/machinery/power/rad_collector
+	name = "radiation collector array"
+	desc = "A device which uses radiation and phoron to produce power."
+	icon = 'icons/obj/machinery/rad_collector.dmi'
+	icon_state = "ca"
+	anchored = FALSE
+	density = TRUE
+	req_access = list(ACCESS_ENGINE_EQUIP)
+	var/obj/item/tank/phoron/P = null
+
+	var/health = 100
+	var/max_safe_temp = 1000 + T0C
+	var/melted
+
+	var/last_power = 0
+	var/last_power_new = 0
+	var/active = 0
+	var/locked = 0
+	var/drainratio = 1
+
+	var/last_rads
+	var/max_rads = 250 // rad collector will reach max power output at this value, and break at twice this value
+	var/max_power = 5e5
+	var/pulse_coeff = 20
+	var/end_time = 0
+	var/alert_delay = 10 SECONDS
+
+/obj/machinery/power/rad_collector/Initialize()
+	. = ..()
+	rad_collectors += src
+
+/obj/machinery/power/rad_collector/Destroy()
+	rad_collectors -= src
+	return ..()
+
+/obj/machinery/power/rad_collector/process()
```

Comment:
Make this proc time constant, you receive a seconds_per_tick parameter and everything relevant have to be multiplied by it, as outlined in the proc header